tower-hedge/src/delay/error.rs

```diff
@@ -0,0 +1,27 @@
+//! Error types
+
+use std::fmt;
+
+#[derive(Debug)]
+pub enum Error {
+    TimerError(tokio_timer::Error),
+    ServiceError(::Error),
+}
+
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            Error::TimerError(e) => Some(e),
+            Error::ServiceError(e) => Some(&**e),
+        }
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::TimerError(e) => write!(f, "timer error in delay: {:?}", e),
```

Comment:
The `Display` implementation should not forward to the inner error if `source` also returns the inner error. The user is expected to iterate the `source` chain if they want the error message to be of the format "top level error: lower level cause".